java/src/main/java/org/rocksdb/RocksIterator.java

```diff
@@ -89,9 +96,16 @@ public int key(ByteBuffer key) {
    *     input buffer {@code value} is insufficient and partial result will
    *     be returned.
    */
-  public int value(ByteBuffer value) {
-    assert (isOwningHandle() && value.isDirect());
-    int result = valueDirect0(nativeHandle_, value, value.position(), value.remaining());
+  public int value(final ByteBuffer value) {
```

Comment:
do we plan to support more methods in the future? `RocksDB.get/put/delete` and `WriteBatch.put/delete`

Reply:
Yes, but that will be done in separate PRs